src/datagrid/datagrid.component.jsx

```diff
@@ -132,7 +132,7 @@ export default class DataGrid extends React.PureComponent {
         case KEY_CODES.RIGHT:
         case KEY_CODES.LEFT: {
           e.preventDefault();
-          let columnInd = columns.findIndex(c => c.valueKeyPath.join() === columnKey);
+          let columnInd = columns.findIndex(c => c.valueKeyPath.join('/') === columnKey);
```

Comment:
The '/' separator is needed here because getColumnKey in Utils also uses '/' separator and and the columnKey is generated using Utils.getColumnKey

const getColumnKey = col => (
  col.columnKey || col.valueKeyPath.join('/')
);